dafny-lang/dafny
# Out of resources error reporting

### Description
- When reporting a verification issue other than an assertion failure, such as running out of resources, use the token of the assertion batch, not of the entire verifiable symbol. This allows the user to figure out where the bad assertion is, and prevents duplicate error messages.
- Report the result of verification when using the `--progress` option
- Tweak the formatting of `--progress`

#### Examples
```dafny
lemma{:resource_limit 10000000} L()
{
  assert true;
  assert f(10, 5) == 0; // runs out of resources
  assert true;
  assert f(10, 6) == 0; // runs out of resources
}
```

Results in 
```
outOfResourceAndIsolateAssertions.dfy(4,18): Error: Verification out of resource (L)
outOfResourceAndIsolateAssertions.dfy(6,18): Error: Verification out of resource (L)
```

New look of `--progress`:
```
Verification part 7/9 of L, on line 6, verified successfully, taking 32ms and consuming 6.8E+003 resources
Verification part 8/9 of L, on line 4, ran out of resources, taking 23ms and consuming 1.0E+007 resources
Verification part 9/9 of L, on line 6, ran out of resources, taking 48ms and consuming 1.0E+007 resources
```

### How has this been tested?
- Add a CLI test `outOfResourceAndIsolateAssertions.dfy` 
- Update existing tests using --progress

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>
